wasmplugin/plugin.go

```diff
@@ -357,20 +358,27 @@ func (ctx *httpContext) OnHttpStreamDone() {
 	logMemStats()
 }
 
-func (ctx *httpContext) handleInterruption(phase string, interruption *ctypes.Interruption) types.Action {
-	ctx.metrics.CountTXInterruption(phase, interruption.RuleID)
+const noGRPCStream int32 = -1
 
-	proxywasm.LogInfof("%d interrupted, action %q", ctx.contextID, interruption.Action)
-	statusCode := interruption.Status
-	if statusCode == 0 {
-		statusCode = 403
-	}
+func (ctx *httpContext) handleInterruption(phase string, interruption *ctypes.Interruption) types.Action {
+	if !ctx.triggeredInterruption {
```

Comment:
I guess here we ensure idempotency, I wonder in which cases the handleInterruption is called twice. Could you dig more into it e.g. when request is X and rule is Y.